source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -778,29 +778,18 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
                     }
                 }
 
-                bool addedDefault = false;
                 if (nodedefPort->isA<Input>())
                 {
                     GeomPropDefPtr geomprop = nodedefPort->asA<Input>()->getDefaultGeomProp();
                     if (geomprop)
                     {
                         inputSocket->setGeomProp(geomprop->getName());
                         input->setGeomProp(geomprop->getName());
-
-                        const string& connection = nodePort ? nodePort->asA<Input>()->getOutputString() : EMPTY_STRING;
-                        if (connection.empty())
-                        {
-                            graph->addDefaultGeomNode(input, *geomprop, context);
-                            addedDefault = true;
-                        }
                     }
                 }
 
                 // Connect to the graph input
-                if (!addedDefault)
-                {
-                    inputSocket->makeConnection(input);
-                }
+                inputSocket->makeConnection(input);
```

Comment:
In this code path we are generating a shader from a single node, and the shader interface should match the node's interface. So always connect all node inputs to the graph interface, and do not create and connect a geomprop node.

Default genomprop data is set anyway when emitting the shader inputs in OSL and MDL.

Reply:
LGTM

Reply:
Just double-checking that:
```
inputSocket->setGeomProp(geomprop->getName());
```
will still expose the stream requirement so can be bound to if found.

Reply:
Yes, this is handled in OSL & MDL where a default "value" is set (where value is the function calculating the default geometric data).

In GLSL we can't do the same of course, but I don't think it makes sense to generate single node function shaders in GLSL anyway.

We might need to take a more holistic look at geomprop generation later. The main thing with this quick fix is that now the shader signature will match the node signature.